src/compiler/interpreter.jl

```diff
@@ -225,25 +359,35 @@ function Core.Compiler.abstract_call_gf_by_type(
     elseif is_alwaysinline_func(specTypes)
         callinfo = AlwaysInlineCallInfo(callinfo, atype)
     else
-        # 1. Check if function is inactive
-        if is_inactive_from_sig(interp, specTypes, sv)
+        method_table = Core.Compiler.method_table(interp)
+        if EnzymeRules.is_inactive_from_sig(specTypes; world = interp.world, method_table)
             callinfo = NoInlineCallInfo(callinfo, atype, :inactive)
         else
-            # 2. Check if rule is defined
-            has_rule = get!(interp.rules_cache, specTypes) do
-                if interp.forward_rules && has_frule_from_sig(interp, specTypes, sv)
```

Comment:
Probably should have then deleted the code in `tfuncs.jl` since this was the sole user.